TLSharp.Core/TelegramClient.cs

```diff
@@ -28,8 +28,10 @@ public class TelegramClient : IDisposable
         private int _apiId = 0;
         private Session _session;
         private List<TLDcOption> dcOptions;
+        private TcpClientConnectionHandler _handler;
 
-        public TelegramClient(int apiId, string apiHash, ISessionStore store = null, string sessionUserId = "session")
+        public TelegramClient(int apiId, string apiHash,
+            ISessionStore store = null, string sessionUserId = "session", TcpClientConnectionHandler handler = null)
```

Comment:
For me, all mandatory parameters are on the 1st

- apiId, apiHash

Optional parameters on a 2nd line

- store, sessionUserId, handler

I might not get what you asked...